snuba/api.py

```diff
@@ -268,36 +268,15 @@ def parse_and_run_query(dataset, request: Request, timer):
     from_date, to_date = TimeSeriesExtensionProcessor.get_time_limit(request.extensions['timeseries'])
 
     extensions = dataset.get_extensions()
+
     for name, extension in extensions.items():
         extension.get_processor().process_query(
             request.query,
             request.extensions[name],
+            request.settings
         )
     request.query.add_conditions(dataset.default_conditions())
 
-    # NOTE: we rely entirely on the schema to make sure that regular snuba
-    # queries are required to send a project_id filter. Some other special
-    # internal query types do not require a project_id filter.
-    project_ids = util.to_list(request.extensions['project']['project'])
-    if project_ids:
-        request.query.add_conditions([('project_id', 'IN', project_ids)])
-
-    turbo = request.extensions['performance'].get('turbo', False)
-    if not turbo:
-        final, exclude_group_ids = get_projects_query_flags(project_ids)
-        if not final and exclude_group_ids:
-            # If the number of groups to exclude exceeds our limit, the query
-            # should just use final instead of the exclusion set.
-            max_group_ids_exclude = state.get_config('max_group_ids_exclude', settings.REPLACER_MAX_GROUP_IDS_TO_EXCLUDE)
-            if len(exclude_group_ids) > max_group_ids_exclude:
-                final = True
-            else:
-                request.query.add_conditions([(['assumeNotNull', ['group_id']], 'NOT IN', exclude_group_ids)])
-    else:
-        final = False
```

Comment:
I think this logic is missing one.
```
If turbo:
    final = False
```
This should be applied no matter what and be independent from the project extension.
I would add it here or when you format the query, since there, you still have access to both settings and query.